### DISCUSSION TITLE
Feature Idea - Multi Camera Detection

### DISCUSSION THREAD

**Discussion by szaske** [blakeblackshear/frigate#7394] [Ideas]
My driveway is viewed by two cameras from opposite sides of the driveway.  I love Frigate, but I do still get false positives. One feature I'd like to see in Frigate is some way to have:

1. Frigate be smart enough to understand that these cameras are viewing the same scenes from different angles
2. Be able to reduce false positives, by using that information to gain a higher level of confidence with it's detections.  So if both cameras see a person enter my driveway, that should count as a very high confidence event.

I think this would be very slick and would take Frigate to the next level.  What do you think?

**Comment by Cold-Lemonade** [comment]
I think there's a deeper statistical problem here. There are to types of mistakes that one can make--false positives and false negatives. In your context, a false negative is when someone is on your driveway but is not detected. 

Suppose hypothetically that you could use two cameras to determine that someone was on your driveway. If both cameras have to detect the person, this would certainly reduce the rate of false positives, but it necessarily increases the rate of false negatives. For example, there will be times when someone is actually on your driveway but is missed by at least one camera. Then your hypothetical system would fail to detect them even though someone is there. In statistics, this is known as the inverse relationship between the rates of Type I and Type II errors.

I live in a high crime neighborhood. For me, the risk of a false negative is much worse than a false positive. I would rather be notified that someone is there when there is no one there (false positive), then miss someone when there is actually someone on my property (false negative). There are areas of my property covered by two cameras and I have set Frigate up to notify me if either camera detects a person. So I get more false positives and few false negatives.

**Comment by NickM-27** [comment]
I don't think this approach makes sense for false positive handling. The answer to that (IMO) is better models like what frigate+ is going to be offering. 

I think this feature would make sense though for keeping a single event running as an object is tracked across multiple cameras.

**Reply by OnFreund** [reply]
I think this approach can be useful for false positive handling if you two cameras let you use zones in a way that one doesn't. For example, if one camera covers the parking area and nothing else, and the other can capture the driveway leading to it, and then you can have the zone from camera #2 be required for an event in camera #1.

**Reply by NickM-27** [reply]
what you're describing doesn't sound like false positive (an object that is detected as something when it is not that thing) hanlding, but just a feature request

**Reply by OnFreund** [reply]
You're right, I'll be a bit more specific - it can eliminate false events from stationary objects being detected again due to undesired motion / lighting changes / being partially covered, when the camera covering the stationary part can't be divided into zones.
